src/master-playlist-controller.js

```diff
@@ -354,18 +354,18 @@ export class MasterPlaylistController extends videojs.EventTarget {
       new SegmentLoader(videojs.mergeOptions(segmentLoaderOptions, {
         segmentMetadataTrack: this.segmentMetadataTrack_,
         loaderType: 'main'
-      }));
+      }), options);
```

Comment:
Why not just merge `options` into the first parameter instead of having two parameters that are both option objects?